app/services/foreman_templates/template_exporter.rb

```diff
@@ -0,0 +1,92 @@
+module ForemanTemplates
+  class TemplateExporter < Action
+    def self.setting_overrides
+      super + %i(metadata_export_mode)
+    end
+
+    def export!
+      if git_repo?
+        export_to_git
+      else
+        export_to_files
+      end
+
+      return true
+    end
+
+    def export_to_files
+      @dir = get_absolute_repo_path
+      verify_path!(@dir)
+      dump_files!
+    end
+
+    def export_to_git
+      @dir = Dir.mktmpdir
+
+      git_repo = Git.clone(@repo, @dir)
+      logger.debug "cloned #{@repo} to #{@dir}"
+      branch = @branch ? @branch : get_default_branch(git_repo)
+      git_repo.checkout(branch) if branch
```

Comment:
Is there any chance to create the branch if we specify a non-existant one? Currently it fails unless you give a valid branch, and it felt a bit weird to have to clone my repo separately just to add a target for exporting. This is not a blocker and can be done in another PR, just curious